org.civicrm.reqangular/src/common/services/api.js

```diff
@@ -121,9 +122,13 @@ define([
             action: action
           }
         }).then(function (response) {
-
-          return response.data;
+          if(!!response.data.is_error) {
+            deferred.reject(response.data.error_message);
+          }
+          deferred.resolve(response.data);
         });
+
+        return deferred.promise;
       },
```

Comment:
you are not returning $http, you are only resolving/rejecting in then call, so how will clients of the call know when $http itself throws error